contracts/HtsSystemContract.sol

```diff
@@ -48,6 +49,265 @@ contract HtsSystemContract is IHederaTokenService, IERC20Events, IERC721Events {
         (responseCode, tokenInfo) = IHederaTokenService(token).getTokenInfo(token);
     }
 
+    function getNonFungibleTokenInfo(address token, int64 serialNumber)
+        htsCall external
+        returns (int64, NonFungibleTokenInfo memory) {
+        require(token != address(0), "getNonFungibleTokenInfo: invalid token");
+
+        (int64 responseCode, TokenInfo memory tokenInfo) = IHederaTokenService(token).getTokenInfo(token);
+        require(responseCode == 22, "getNonFungibleTokenInfo: failed to get token data");
```

Comment:
let's extract this to a library like https://github.com/hashgraph/hedera-smart-contracts/blob/ffe75e67cbf4011b561be571b5747251cbb60dad/contracts/system-contracts/HederaResponseCodes.sol#L31 with only `HederaResponseCodes.SUCCESS` and replace any other usage